cpp_guidelines/.clang-format

```diff
@@ -0,0 +1,107 @@
+---
+Language: Cpp
+Standard: c++17
+DisableFormat:   false
+AccessModifierOffset: -4
+AlignAfterOpenBracket: Align
+AlignConsecutiveAssignments: true
+AlignConsecutiveBitFields: true
+AlignConsecutiveDeclarations: false
+AlignConsecutiveMacros: true
+AlignEscapedNewlines: Right
+AlignOperands: true
+AlignTrailingComments: true
+AllowAllArgumentsOnNextLine: false
+AllowAllConstructorInitializersOnNextLine: false
+AllowAllParametersOfDeclarationOnNextLine: false
+AllowShortBlocksOnASingleLine: Never
+AllowShortCaseLabelsOnASingleLine: false
+AllowShortEnumsOnASingleLine: false
+AllowShortFunctionsOnASingleLine: None
+AllowShortIfStatementsOnASingleLine: Never
+AllowShortLambdasOnASingleLine: All
+AllowShortLoopsOnASingleLine: false
+AlwaysBreakAfterReturnType: None
+AlwaysBreakBeforeMultilineStrings: false
+AlwaysBreakTemplateDeclarations: Yes
+BinPackArguments: false
+BinPackParameters: false
+BitFieldColonSpacing: Both
+BreakBeforeBinaryOperators: All
+BreakBeforeBraces: Allman
+BreakBeforeTernaryOperators: true
+BreakConstructorInitializers: BeforeComma
+BreakInheritanceList: BeforeComma
+BreakStringLiterals: true
+ColumnLimit:     79
+CommentPragmas:  '^ COMMENT pragma:'
+CompactNamespaces: false
+ConstructorInitializerAllOnOneLineOrOnePerLine: true
+ConstructorInitializerIndentWidth: 4
+ContinuationIndentWidth: 4
+Cpp11BracedListStyle: true
+DeriveLineEnding: false
+DerivePointerAlignment: false
+ExperimentalAutoDetectBinPacking: false
+FixNamespaceComments: true
```

Comment:
I dislike this, because it just adds more clutter and it is rarely helping me. But I have no strong opinions on this. clang-format adds those automatically for me, so at least I do not need to type them.

Reply:
IMO this comments are very useful. For example in alpaka we do not use namespace comments and it is very hard to found where the code corresponds to.

Reply:
I am also much more often distracted by the noise which those comments add. If I want to find out the namespace of an object, the IDE will tell me easily.
Furthermore, this option is buggy and sometimes misses the comment for the innermost namespace.

Reply:
**VOTE**
current | false |
--------|-------------|
 0      |   0.5         |

Namespace comments add more noise to the program. But they might be helpful when viewing a file outside a good text editor or IDE.

Reply:
current | false
-- | --
1 | 0.5

I am working with online viewer e.g. github diffs and sometimes with plain vim via ssh on different HPC systems, those comments IMO help's a lot in alpaka and picongpu.

Reply:
current | false |
----------|--------|
2 | 0.5 |

Reply:
current | false |
--------|-----------|
 3      |   0.5

Reply:
current | false
-- | --
3 | 1.5

Reply:
current | false |
--------|-----------|
 4      |   1.5

Reply:
current | false
-- | --
5 | 1.5

Reply:
current | false
-- | --
5 | 2.5

Reply:
Reminder: Voting will be closed on 2020-09-26 at 12:28 HZDR time.

Reply:
Voting closed, the winner is `true`.